Integrated Gameplay Systems (Prototype)/Assets/Scripts/BaseBuilding/DemolishMenu.cs

```diff
@@ -3,20 +3,34 @@
 using UnityEngine;
 using System;
 
-public class DemolishMenu : Menu
+public class DemolishMenu : Menu, ICommand
 {
+    private GameObject playerCamera;
     private OpenMenuCommand backCommand;
 
     public DemolishMenu(IStateMachine _stateMachine, GameManager _gameManager) : base(_stateMachine, _gameManager) {
         allowMovement = true;
         backCommand = new OpenMenuCommand(typeof(BuildMenu), _stateMachine, _gameManager);
+
+        playerCamera = (GameObject) gameManager.GetObjectWithTag("Camera");
     }
 
     public override void EnableState() {
         gameManager.inputManager.RegisterKeyBinding(KeyCode.Escape, backCommand);
+        gameManager.inputManager.RegisterKeyBinding(KeyCode.Mouse0, this);
     }
 
     public override void DisableState() {
         gameManager.inputManager.DeregisterKeyBinding(KeyCode.Escape, backCommand);
+        gameManager.inputManager.DeregisterKeyBinding(KeyCode.Mouse0, this);
```

Comment:
Ik zie dat je dezelfde toets gebruikt voor het openen van een menu en het sluiten. Toen ik nog mijn bug had dat ik meerdere inventory menu's maakte, merkte ik echter dat het menu constant sluit en opent, omdat de inputHandler waarschijnlijk GetKey doet ipv GetKeyDown. Dus als je hier problemen mee ervaart moeten we dat even oplossen